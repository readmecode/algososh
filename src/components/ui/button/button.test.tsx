import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("test uiButton", () => {
  test("with text", () => {
    const tree = renderer.create(<Button text={"button text"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("locked", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with loader", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("processing a callback in a button", () => {
    window.alert = jest.fn();

    const callback = () => {
      alert("callback");
    };

    // Рендерим компонент
    render(<Button text={"button text"} onClick={callback} />);

    // Находим элемент
    const button = screen.getByText("button text");

    // Имитируем нажатие
    fireEvent.click(button);

    expect(window.alert).toHaveBeenCalledWith("callback");
  });
});
