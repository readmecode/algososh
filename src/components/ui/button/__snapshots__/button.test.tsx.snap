// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test uiButton locked 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`test uiButton with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`test uiButton with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    button text
  </p>
</button>
`;

exports[`test uiButton without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
